### PR TITLE
JSM: Added module and TS file for ConvexObjectBreaker

### PR DESCRIPTION
**Pull request by yomboprime**
Module generated by `modularize.js`

I've tested that the example works OK with modules, in another directory.

I've not tested the TS file (and it is manually created)

Edit:
To be merged after #16697
Related #16688

### CODE REVIEW

File: examples/jsm/misc/ConvexObjectBreaker.js [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+		resultPlane.constant = - referencePoint.dot( resultPlane.normal );
+
+
+	};
+
+}();
+
+export { ConvexObjectBreaker };
+
+export {  };
```
Comment (yomboprime) [inline examples/jsm/misc/ConvexObjectBreaker.js]: Is it OK this second `export` keyword or it shouldn't be here?

Comment (Mugen87) [reply examples/jsm/misc/ConvexObjectBreaker.js]: No, the second export should no be there. I've created the module on my computer and I only have a single export statement...

**Comment by Mugen87** [comment]
You also have to move the original `ConvexObjectBreaker.js` to the `misc` directory (and adjust the respective example). Otherwise executing `modularize.js` in the repository won't work.

File: examples/jsm/misc/ConvexObjectBreaker.js [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+ * @param {double} smallDelta Max distance to consider that a point belongs to a plane.
+ *
+*/
+
+
+import { ConvexBufferGeometry } from "../geometries/ConvexGeometry.js";
+
+import {
+	Line3,
+	Matrix4,
```
Comment (Mugen87) [inline examples/jsm/misc/ConvexObjectBreaker.js]: Importing `Matrix4` is not necessary. You can put it on the ignore list in `modularize.js`.

File: examples/jsm/misc/ConvexObjectBreaker.js [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+ * Note: This lib adds member variables to object's userData member (see prepareBreakableObject function)
+ * Use with caution and read the code when using with other libs.
+ *
+ * @param {double} minSizeForBreak Min size a debris can have to break.
+ * @param {double} smallDelta Max distance to consider that a point belongs to a plane.
+ *
+*/
+
+
+import { ConvexBufferGeometry } from "../geometries/ConvexGeometry.js";
```
Comment (Mugen87) [inline examples/jsm/misc/ConvexObjectBreaker.js]: Um, this  import statement should actually be the second one. I wonder why it is located at first place.

**Comment by Mugen87** [comment]
The previous merges produces a conflict. Do you mind resolving^^?

File: examples/jsm/misc/ConvexObjectBreaker.js (hunk 1 above)
Comment (yomboprime) [reply examples/jsm/misc/ConvexObjectBreaker.js]: Ok, i was modularizing an already modularized .js, that's why 😜

Now it is right.

File: examples/jsm/misc/ConvexObjectBreaker.js (hunk 2 above)
Comment (yomboprime) [reply examples/jsm/misc/ConvexObjectBreaker.js]: Done.

File: examples/jsm/misc/ConvexObjectBreaker.js (hunk 3 above)
Comment (yomboprime) [reply examples/jsm/misc/ConvexObjectBreaker.js]: It was because of the same cause as the second import. Now it is right.

**Comment by yomboprime** [comment]
I've not made a resolve before. I know how to edit the conflicted file but no idea of the necessary git commands.

Git instructs me to do this:

```
Step 1: From your project repository, check out a new branch and test the changes.

git checkout -b yomboprime-jsm_convexbreaker dev
git pull https://github.com/yomboprime/three.js.git jsm_convexbreaker

Step 2: Merge the changes and update on GitHub.

git checkout dev
git merge --no-ff yomboprime-jsm_convexbreaker
git push origin dev
```
But I'm not sure of the final `git push origin dev`, because I think it should be pushed on my `jsm_convexbreaker` branch.
Could you help me?

**Comment by looeee** [comment]
For a simple conflict like this you should be able to do it here. Just click the 'resolve conflicts' button, edit the file until it looks right (remove all the >>>> lines) and then mark it resolved.

**Comment by yomboprime** [comment]
> For a simple conflict like this you should be able to do it here. Just click the 'resolve conflicts' button, edit the file until it looks right (remove all the >>>> lines) and then mark it resolved.

Yes, but if I knew how to do it in the console I could resolve more complex conflicts in the future, that's why I asked...

**Comment by yomboprime** [comment]
It seems I did the merge and the resolving right, in the console. Thank you :-)

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**